src/java/htsjdk/samtools/CRAMIndexer.java

```diff
@@ -350,28 +412,30 @@ void startNewReference() {
     public static void createIndex(final SeekableStream stream, final File output, final Log log) throws IOException {
 
         final CramHeader cramHeader = CramIO.readCramHeader(stream);
+        if (cramHeader.getSamFileHeader().getSortOrder() != SAMFileHeader.SortOrder.coordinate) {
+            throw new SAMException("Expecting a coordinate sorted file.");
+        }
         final CRAMIndexer indexer = new CRAMIndexer(output, cramHeader.getSamFileHeader());
 
         int totalRecords = 0;
         Container container = null;
         do {
-            if (++totalRecords % 10 == 0)
-                if (null != log) log.info(totalRecords + " slices processed ...");
+            if (++totalRecords % 10 == 0) {
+                if (null != log) {
+                    log.info(totalRecords + " slices processed ...");
+                }

```

Comment:
Use a [SAMProgressLogger](https://github.com/samtools/htsjdk/blob/e64903b5e977a6ee88748ee71581db341758d457/src/java/htsjdk/samtools/util/ProgressLogger.java#L20) for this.